### DISCUSSION TITLE
AI Deck GAP8 Development environment setup: Make autotiler --> Critical Error

### DISCUSSION THREAD

**Discussion by LucasTheIntern** [bitcraze/discussions#1096] [Q&A] [answered]
Hi there, 

I'm trying to follow the instructions for setting up a development environment for the GAP8 chip found in the "Getting started with the AI deck" page:
https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/

Under the "Setting up docker and the autotiler" heading near the end of the page, there are a few terminal commands:
`$ docker run --rm -it --name myAiDeckContainer bitcraze/aideck`
`$ cd /gap_sdk`
`$ source configs/ai_deck.sh`
`$ make autotiler`

All goes well until the last one. It begins to execute normally (I guess), but then suddenly freezes, and a VirtualBox popup appears stating that a critical error has occurred:
![Kaput](https://github.com/bitcraze/discussions/assets/120214901/4a066297-baf1-4059-b104-86325d9804ad)

Here's a closer look at the frozen terminal output:
![terminal](https://github.com/bitcraze/discussions/assets/120214901/c22f4d69-0c79-495d-bea3-d79ac069e768)
Unfortunately, I cannot copy/paste all of the output since the VM will not allow any interaction. 

The .png file referenced by the popup is the same screenshot I showed above. I opened the log file... it doesn't appear to contain anything very enlightening. The last few lines seem to be showing some kind of port checks, and then the last line is just a bunch of exclamation marks:
`00:07:07.072364   Flags 3 ( IS_PRIMARY ), not modified`
`00:07:07.072367 !!`
`00:07:07.072367 !! {vmsvga3dctx}`
`00:07:07.072368 !!`
`00:07:07.072372 !!`
`00:07:07.072372 !! {vmsvga3dsfc}`
`00:07:07.072373 !!`
`00:07:07.072379 !!`
`00:07:07.072379 !! {vmsvga3dsurf}`
`00:07:07.072380 !!`
`00:07:07.072381 !!`
`00:07:07.072381 !! {xhci}`
`00:07:07.072382 !!`
`00:07:07.072386 usb-xhci#0: PCI MMIO=00000000f0810000 IRQ=11 MSI=off R0=true RC=false`
`00:07:07.072389 USBCMD: 5: INTE RS`
`00:07:07.072391 USBSTS: 0:`
`00:07:07.072392 DNCTRL: 2   CONFIG: 20 (32 slots)`
`00:07:07.072393 DCBAA ptr: 0000000001405000`
`00:07:07.072395 CRCR: 1: RCS`
`00:07:07.072396 CRCR ptr : 0000000001406000`
`00:07:07.072397 Port 01 (USB2): PORTSC:  A0002A0: PLS=5 PP SPD=0 WCE WOE`
`00:07:07.072403 Port 02 (USB2): PORTSC:  A0002A0: PLS=5 PP SPD=0 WCE WOE`
`00:07:07.072407 Port 03 (USB2): PORTSC:  A0002A0: PLS=5 PP SPD=0 WCE WOE`
`00:07:07.072410 Port 04 (USB2): PORTSC:  A0002A0: PLS=5 PP SPD=0 WCE WOE`
`00:07:07.072413 Port 05 (USB2): PORTSC:  A0002A0: PLS=5 PP SPD=0 WCE WOE`
`00:07:07.072416 Port 06 (USB2): PORTSC:  A0002A0: PLS=5 PP SPD=0 WCE WOE`
`00:07:07.072419 Port 07 (USB2): PORTSC:  A0002A0: PLS=5 PP SPD=0 WCE WOE`
`00:07:07.072422 Port 08 (USB2): PORTSC:  A0002A0: PLS=5 PP SPD=0 WCE WOE`
`00:07:07.072424 Port 09 (USB3): PORTSC:  A0002A0: PLS=5 PP SPD=0 WCE WOE`
`00:07:07.072427 Port 10 (USB3): PORTSC:  A0002A0: PLS=5 PP SPD=0 WCE WOE`
`00:07:07.072430 Port 11 (USB3): PORTSC:  A0002A0: PLS=5 PP SPD=0 WCE WOE`
`00:07:07.072433 Port 12 (USB3): PORTSC:  A0002A0: PLS=5 PP SPD=0 WCE WOE`
`00:07:07.072436 Port 13 (USB3): PORTSC:  A0002A0: PLS=5 PP SPD=0 WCE WOE`
`00:07:07.072439 Port 14 (USB3): PORTSC:  A0002A0: PLS=5 PP SPD=0 WCE WOE`
`00:07:07.072442 !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!`

Here is the log file in its entirety:
[VBox.log](https://github.com/bitcraze/discussions/files/13730529/VBox.log)

I am able to run all of these commands again after restarting the VM, With exactly the same results.

**Comment by ataffanel** [comment, marked as answer]
Thanks for the detailed report! This is the first time I see the VM crashing virtualbox, so it is an odd one.

From the log I could figure that the original problem seems to be `!!R0-Assertion Failed!!`. A quick search shows that enabling `Nested Paging` could solve the problem: https://forums.virtualbox.org/viewtopic.php?p=537277&sid=5a260d559ce05e8e88e9b5c72006dbf3#p537277. This is something you could test.

**Reply by LucasTheIntern** [reply]
This worked! Thank you!

For anyone wondering _how_ to enable nested paging, 
1. Power off/send shutdown signal to the VM.
2. In the virtualbox VM manager, select the VM and navigate to settings>system>acceleration and check the box labeled "Hardware Virtualization: Enable Nested Paging".
3. Boot up VM!
